### PR TITLE
[ANCHOR-764] Add callback_urls to client config

### PR DESCRIPTION
**Pull request by philipliu**
### Description

The `callback_urls` parameter has been added to 2.9. This needs to be adapted for the 3.0 client config format.

### Context

This is blocking a merge from develop.

### Testing

- `./gradlew test`

### Documentation

Client config will need to be updated in 3.0.

### Known limitations

N/A

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/client/ClientConfig.java
Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/client/ClientConfig.java:16]: > How does the handler uses the newly added `callback_urls`?

We don't use it right now. `develop` currently uses the old client config schema, when I merge `develop` into `feature/v3`, the implementation will be updated. I published this PR to reduce the implementation work needed for the merge.

> Can we also reflect this change in the` anchor_platform_default_values.yaml`?

The client configs no longer live in the anchor config. There is no change required to `anchor_platform_default_values`.

Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/client/ClientConfig.java:16]: When we add fields to the config, it helps to add comments to the config.

**Comment by lijamie98** [comment]
Is there a reason to have separate PRs? 
https://github.com/stellar/java-stellar-anchor-sdk/pull/1465

File: core/src/main/java/org/stellar/anchor/client/ClientConfig.java
Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/client/ClientConfig.java:16]: >  I published this PR to reduce the implementation work needed for the merge.
Can you elaborate a little? Is there a reason that we cannot add implementation in the same PR?

Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/client/ClientConfig.java:16]: > When we add fields to the config, it helps to add comments to the config.

I just checked this there is a comment block about the client config. I'll add it.

Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/client/ClientConfig.java:16]: >I published this PR to reduce the implementation work needed for the merge.
> Can you elaborate a little? Is there a reason that we cannot add implementation in the same PR?

The existing implementation in `develop` uses the old client config format. The callback URLs are flat strings rather than nested in a `callbackUrls` object. To merge `develop` into `feature/v3`, I will need to move this implementation into the new client config format. This means we need to implement the changes in this PR. I think it's better to keep the merge PR light in additional implementation.

**Comment by philipliu** [comment]
> Is there a reason to have separate PRs?

This PR adds the callback URL configuration in the new client config schema. The other PR adds validation and tests for the 2.x client config format.

File: core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt
Comment (JiahuiWho) [inline core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt:154]: How about migrate to use `callbackUrls` ince the `callbackUrl` is already marked as deprecated?

Comment (philipliu) [reply core/src/test/kotlin/org/stellar/anchor/sep24/Sep24ServiceTest.kt:154]: I plan to migrate the usage in the merge PR

**Review by JiahuiWho — APPROVED** [review]
lgtm

File: core/src/main/java/org/stellar/anchor/client/ClientConfig.java
Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/client/ClientConfig.java:16]: We are planning to switch the clients config back to a flatter format. Maybe it would be a good idea to merge this PR after the refactoring?

Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/client/ClientConfig.java:16]: This shouldn't be majorly impacted by the refactoring and is currently blocking merging `develop` into `feature/v3`.

**Review by lijamie98 — APPROVED** [review]

**Pull request merged**
